### ISSUE TITLE
Waiting on multiple condition does not unblock even after one of the condition is met

### ISSUE THREAD

**Issue by yashgorana** [kr8s-org/kr8s#292]
### Which project are you reporting a bug for?

kr8s

### What happened?

When waiting on multiple conditions, job.wait() does NOT unblock if any one of them match correctly.

#### Repro code
```python
import kr8s
from kr8s.objects import Job

client = kr8s.api(namespace="syft")
job = Job(
    {
        "metadata": {
            "name": "test-job",
        },
        "spec": {
            "backoffLimit": 0,
            "ttlSecondsAfterFinished": 60,
            "template": {
                "spec": {
                    "containers": [
                        {
                            "name": "python",
                            "image": "python:3.11-alpine",
                            "command": [
                                "python",
                                "-c",
                                "raise Exception('mock exception')",
                            ],
                        }
                    ],
                    "restartPolicy": "Never",
                }
            },
        },
    }
)
job.create()
job.wait(["condition=Complete", "condition=Failed"])
print(job.raw)
```

#### Expected Result:
- Should unblock when `Complete` or `Failed` matches the provided value or `True`

#### Actual Result
-  Unblocks only when the first condition in the list matches the expected value
- Ignores any subsequent conditions and remains blocked forever.

### Anything else?

The issue lies in [`_test_conditions()`](https://github.com/kr8s-org/kr8s/blob/main/kr8s/_objects.py#L380-L381). If first condition fails the value check, the loop instantly terminates with a return value of `False` and the subsequent conditions are never checked, and hence `wait()` is never unblocked.

**Comment by jacobtomlinson** [comment]
Thanks for raising this @yashgorana.

The way I've treated the implementation of waiting on multiple conditions in kr8s is as an `AND`, hence why it is returning `False` as soon as it finds a condition that is not met. I hadn't considered the case where you pass the same condition with different values, which implies an `OR`. 

Interestingly it seems like `kubectl` doesn't implement this at all currently and only supports waiting on a single condition, so there is no reference implementation to inform the correct behaviour here. 

It looks like kubernetes/kubernetes#95759 is tracking adding support for this. There is a proposed implementation in kubernetes/kubernetes#119205 which would make conditions passed with `--for` (or the `for` kwarg in `kr8s`) an `OR` and introduces a new `--for-all` (which we would implement in `kr8s` as `for_all`) as an `AND`.

Ideally I would like to wait for that PR to make it into a release before changing behaviour here as a core goal for `kr8s` is to behave the same way `kubectl` does. To take that to an extreme we could update the implementation here so that `for` only takes a single condition to match the current `kubectl` behaviour.

Another option could be to update the logic here to make it an `OR` and then when that PR lands we could add the `for_all` later to reinstate the `AND`.

A third option could be to expose the `_test_conditions()` method as a public method to make it easier to implement whatever behaviour you want on the user side, although this is a slight addition to the `kubectl` API.

**Comment by jacobtomlinson** [comment]
In the meantime you could work around this using timeouts and checking both conditions in a loop. It's a bit gross but if you check each condition with a short timeout and break if the condition is met while suppressing the exception it will check each condition each second and break out of the loop if one of the conditions is met.

```python
from contextlib import suppress

while True:
    with suppress(TimeoutError):
        job.wait(["condition=Complete"], timeout=0.5)
        break
    with suppress(TimeoutError):
        job.wait(["condition=Failed"], timeout=0.5)
        break
```

**Comment by jacobtomlinson** [comment]
Ok decided to go ahead and implement the proposed kubectl functionality. I've just released `v0.13.0` which makes your initial example work as expected.

```python
from kr8s.objects import Job

job = Job(...)
job.create()
job.wait(["condition=Complete", "condition=Failed"])  # Will continue if either condition is true
print(job.raw)
```
